src/main/kotlin/simplerag/ragback/domain/document/service/DataFileService.kt

```
@@ -0,0 +1,108 @@
+            val type = file.resolveContentType()
+
+            if (dataFileRepository.existsBySha256(sha256)) {
+                throw FileException(ErrorCode.ALREADY_FILE, sha256)
+            }
+
+            val fileUrl = s3Util.upload(file, S3Type.ORIGINAL_FILE)
+            uploadedUrls += fileUrl
+
+            val dataFile = try {
+                dataFileRepository.save(DataFile(meta.title, type, sizeByte, sha256, fileUrl, now, now))
+            } catch (ex: DataIntegrityViolationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)